### DISCUSSION TITLE
authelia behind haproxy - TLS between them failed

### DISCUSSION THREAD

**Discussion by mamema** [authelia/authelia#6674] [Q&A] [answered]
Hi,

i have a working config with http and woud like to switch to internal https between authelia and haproxy
but after enabling certs on authelia side, i get any infinity login loop and this in the logs

time="2024-02-10T08:48:11+01:00" level=info msg="Initializing server for TLS connections on '[::]:9091' path '/'"
time="2024-02-10T08:48:13+01:00" level=debug msg="error when serving connection \"192.168.0.xx:9091\"<->\"192.168.0.xx:62672\": tls: first record does not look like a TLS handshake"

i have a wildcard cert btw.

i've stumbled upon this entry: https://github.com/authelia/authelia/issues/877

where a similar issue with traefik occurs.

do i have the same? Does anyone has a working config with haproxy and internal TLS?

**Comment by james-d-elliott** [comment]
The error occurs when the proxy is not attempting to make a TLS handshake (i.e. it's attempting to perform a plain-text HTTP connection) with authelia but you've configured the port to be TLS. The Traefik issue is solved, as long as you configure it correctly and configure the proxy to trust the configured certificate it will work fine. In your case you'll have to configure the lua script correctly.

**Comment by mamema** [comment, marked as answer]
ahh, the lua script, never thought on this... thanks look into it now.

that was nice. two changes http:// to https:// and it works. What a rarely happening thing in information technology.
Thanks!
